qualang_tools/plot/Fitting/fitting.py

```diff
@@ -663,26 +671,27 @@ def reflection_resonator_spectroscopy(
         if guess is not None:
             for key in guess.keys():
                 if key == "f":
-                    f0 = guess[key] / x_normal
+                    f0 = float(guess[key]) / x_normal
                 elif key == "k":
-                    width0 = guess[key] / x_normal
+                    width0 = float(guess[key]) / x_normal
                 elif key == "offset":
-                    v0 = guess[key] / y_normal
+                    v0 = float(guess[key]) / y_normal
                 elif key == "slope":
-                    m = guess[key] / y_normal * x_normal
+                    m = float(guess[key]) / y_normal * x_normal
                 else:
                     raise Exception(
                         f"The key '{key}' specified in 'guess' does not match a fitting parameters for this function."
                     )
 
+        # Print the initial guess if verbose=True
         if verbose:
             print(
-                f"Initial guess:\n"
-                f" f = {f0 * x_normal}, \n"
-                f" kc = {(peak - v0) / width0 * y_normal / x_normal}, \n"
-                f" k = {width0 * x_normal}, \n"
-                f" offset = {v0 * y_normal}, \n"
-                f" slope = {m * y_normal / x_normal}"
+                f"Initial guess:\n "
+                f" f = {f0 * x_normal}, \n "
+                f" kc = {(v0 - peak) * (width0 * x_normal) * y_normal}, \n "
```

Comment:
same here `{(peak - v0) / (width0 * x_normal) * y_normal}` ?

Reply:
same... you can check it and let me know if I  am mistaken ...